controllers/ovs/flowsconfig_types.go

```diff
@@ -0,0 +1,66 @@
+package ovs
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/netobserv/network-observability-operator/api/v1alpha1"
+)
+
+type flowsConfig struct {
+	v1alpha1.FlowCollectorIPFIX `json:",inline"`
+	SharedTarget                string `json:"sharedTarget,omitempty"`
+	NodePort                    int32  `json:"nodePort,omitempty"`
+}
+
+// TODO: for better maintainability, implement map-to-struct/struct-to-map using reflection
+func configFromMap(data map[string]string) (*flowsConfig, error) {
+	config := flowsConfig{
+		FlowCollectorIPFIX: v1alpha1.FlowCollectorIPFIX{
+			CacheActiveTimeout: data["cacheActiveTimeout"],
+		},
+		SharedTarget: data["sharedTarget"],
+	}
+	if val, ok := data["cacheMaxFlows"]; ok {
+		if mf, err := strconv.ParseInt(val, 10, 32); err != nil {
+			return nil, fmt.Errorf("wrong cacheMaxFlows value: %w", err)
+		} else {
+			config.CacheMaxFlows = int32(mf)
+		}
+	}
+	if val, ok := data["sampling"]; ok {
+		if mf, err := strconv.ParseInt(val, 10, 32); err != nil {
+			return nil, fmt.Errorf("wrong sampling value: %w", err)
+		} else {
+			config.Sampling = int32(mf)
+		}
+	}
+	if val, ok := data["nodePort"]; ok {
+		if mf, err := strconv.ParseInt(val, 10, 32); err != nil {
+			return nil, fmt.Errorf("wrong nodePort value: %w", err)
+		} else {
+			config.NodePort = int32(mf)
+		}
+	}
+	return &config, nil
+}
+
+func (fc *flowsConfig) asStringMap() map[string]string {
```

Comment:
Some unit tests on conversion functions would be neat ; e.g. to make sure converting back and forth keeps it consistent

Reply:
They are currently tested through the public APIs, as part of the expected behavior of the controller. With the usage of `mapstructure` library, the test coverage is full now.